### PR TITLE
feat(bucket-brigade): surface critic explained-variance + add single-BR A/B probe

### LINKED ISSUE #241: Tooling: critic explained-variance logging + standalone single-best-response A/B harness for bucket-brigade
## Motivation

Investigating why the PPO best-response (BR) doesn't learn on bucket-brigade (#239) is slow and blunt today: the only signal is the per-iteration `br[val= ent=]` line from a full NFSP run, and there is **no standalone single-BR harness** and **no critic explained-variance metric**. This issue builds the tooling that makes the #239 critic-fitting investigation tractable and tight.

## What to build

1. **Critic explained-variance logging.** In the BR PPO update (`src/multi_agent/joint.rs`), compute and log `explained_variance = 1 - Var(returns - values) / Var(returns)` per update (alongside the existing value-loss/entropy diagnostics surfaced by `examples/games/bucket_brigade/train_nfsp.rs:292-314`). EV near 0 (or negative) = critic not fitting; EV → 1 = critic fits. This is the single most informative missing metric for #239.
2. **Standalone single-best-response harness.** A small example (e.g. `examples/games/bucket_brigade/train_br_probe.rs`) that:
   - freezes N−1 opponents (random or a supplied fixed policy) for one cell,
   - trains ONE PPO best-response agent for K iterations,
   - logs per-iteration: value loss, **explained variance**, policy entropy, mean episode return.
   - is env-var configurable (CELL, ITERATIONS, ROLLOUT_STEPS, the #240 knobs VF_COEF / BR_TRAIN_STEPS / BR_REWARD_SCALE, plus a critic-LR knob if #239 adds one).
   This lets us A/B a single BR in ~1 min instead of via a full NFSP/PSRO run.

## Acceptance criteria

- [ ] Explained-variance is computed and logged in the BR update and surfaced in the bucket-brigade examples.
- [ ] A standalone single-BR harness exists, runs on one cell, and logs val-loss / EV / entropy / mean-return per iteration.
- [ ] Documented one-line command to run a BR A/B.
- [ ] No behavior change to existing trainers (additive logging + a new example); existing tests pass; clippy/fmt clean on both feature sets.

## Context

Part of #239 (PPO BR does not learn). The #240 infrastructure (wired grad-clip, all-minibatch iteration, VF_COEF/BR_TRAIN_STEPS knobs) is already merged. This is the enabler for the remaining #239 investigation.

### PR DESCRIPTION
**Pull request by rjwalters**
## Summary

Tooling for the #239 "PPO best-response does not learn on bucket-brigade" investigation. Two additive deliverables, no behavior change to existing trainers:

1. **Critic explained-variance surfaced.** The BR critic explained-variance (`ev = 1 − Var(returns − values) / Var(returns)`) is already computed inside the joint PPO update (`compute_value_loss` returns it; `JointStats.explained_var` accumulates it per update). This PR surfaces it in the bucket-brigade `train_nfsp` example's per-iteration `br[...]` diagnostic line. EV near 0 / negative = critic not fitting; EV → 1 = fits.

2. **Standalone single-BR A/B probe** (`examples/games/bucket_brigade/train_br_probe.rs`). Freezes N−1 opponents for one cell, trains ONE PPO best-response for K iterations via the joint trainer's freeze-N−1 primitive (`update_with_active_agents`), and logs per-iteration value loss / explained variance / policy entropy / mean episode return. Lets a single BR be A/B'd in ~1 min instead of via a full NFSP/PSRO run.

## Changes

- `examples/games/bucket_brigade/train_nfsp.rs`: add `ev=` to the `br[...]` log line, averaging `JointStats.explained_var` across agents alongside the existing val/ent diagnostics.
- `examples/games/bucket_brigade/train_br_probe.rs` (new): single-BR probe. Env-configurable via `CELL`, `ITERATIONS`, `ROLLOUT_STEPS`, and the #240 knobs `VF_COEF` / `BR_TRAIN_STEPS` / `BR_REWARD_SCALE`. Reuses the existing `collect_rollout` + `update_with_active_agents` plumbing and the PSRO reward-scaling pattern; introduces no new training knobs.
- `Cargo.toml`: register the `train_br_probe` example (`required-features = ["training", "env-bucket-brigade"]`).

## One-line A/B command

```bash
CELL=beta05 ITERATIONS=20 ROLLOUT_STEPS=2048 VF_COEF=0.05 BR_TRAIN_STEPS=4 BR_REWARD_SCALE=0.01 \
  cargo run --release --example train_br_probe --features "training,env-bucket-brigade"
```

Change `VF_COEF` / `BR_TRAIN_STEPS` / `BR_REWARD_SCALE` between two invocations to A/B a single best-response.

## Sample harness output

Captured from an actual run (`CELL=beta05 ITERATIONS=10 ROLLOUT_STEPS=2048 VF_COEF=0.05 BR_TRAIN_STEPS=1 BR_REWARD_SCALE=0.01`):

```
Single-BR probe (issue #241) — Burn backend: NdArray<f32> + Autodiff (CPU)
  cell             = beta05 (β=0.5, κ=0.1, c=0.5)
  br_agent         = 0 (3 opponents frozen)
  iterations       = 10
  rollout_steps    = 2048
  vf_coef          = 0.05 (#240 value-loss weight)
  br_train_steps   = 1 (#240 PPO updates/iter)
  br_reward_scale  = 0.01 (#240/#199 payoff rescale)
  obs_dim          = 47
  action_dims      = [10, 2, 2] (factored multi-discrete)
------------------------------------------------------------
iter   1/10  br[val=1091.3815 ev=-0.0001 ent=1.2274]  mean_ep_return=-23537.56 (16 eps)
iter   2/10  br[val=1192.1418 ev=-0.0000 ent=1.2245]  mean_ep_return=-23755.22 (16 eps)
iter   3/10  br[val=1114.8244 ev=-0.0001 ent=1.2250]  mean_ep_return=-28126.43 (14 eps)
iter   4/10  br[val=1213.8687 ev=-0.0001 ent=1.2194]  mean_ep_return=-26620.27 (15 eps)
iter   5/10  br[val=1280.5556 ev=-0.0000 ent=1.2205]  mean_ep_return=-25378.66 (16 eps)
iter   6/10  br[val=1669.5657 ev=-0.0000 ent=1.2182]  mean_ep_return=-37659.15 (13 eps)
iter   7/10  br[val=1099.2073 ev=-0.0000 ent=1.2136]  mean_ep_return=-22949.88 (17 eps)
iter   8/10  br[val=1314.5082 ev=-0.0000 ent=1.2116]  mean_ep_return=-23613.37 (19 eps)
iter   9/10  br[val=1340.8437 ev=-0.0000 ent=1.2028]  mean_ep_return=-33545.00 (12 eps)
iter  10/10  br[val=1224.3911 ev=-0.0000 ent=1.2123]  mean_ep_return=-29330.18 (14 eps)
------------------------------------------------------------
Probe complete.  iterations=10  time=150.9s
```

Note the `ev` column pinned near 0 — exactly the critic-not-fitting signal the #239 investigation needs to surface and A/B against.

## Acceptance Criteria Verification

| Criterion | Status | Verification |
|-----------|--------|--------------|
| EV computed and logged in the BR update, surfaced in bucket-brigade examples | ✅ | `compute_value_loss`/`JointStats.explained_var` already compute it; `train_nfsp.rs` now prints `ev=` in the per-iter `br[...]` line |
| Standalone single-BR harness, runs on one cell, logs val-loss / EV / entropy / mean-return per iteration | ✅ | New `train_br_probe.rs`; ran it (sample output above) |
| Documented one-line A/B command | ✅ | In this PR body + the example's `//!` doc-comment |
| No behavior change to existing trainers (additive logging + new example) | ✅ | Diff touches only the nfsp example log line, a new example, and Cargo.toml registration |
| Existing tests pass; clippy/fmt clean on both feature sets | ✅ | `cargo test --features "training,env-bucket-brigade"` → 417+ passed, 0 failed; `cargo clippy --all-targets` clean on both `training` and `training,env-bucket-brigade` with `-D warnings`; `cargo fmt --check` clean |

## Test Plan

- `cargo clippy --all-targets --features training -- -D warnings` → exit 0
- `cargo clippy --all-targets --features "training,env-bucket-brigade" -- -D warnings` → exit 0
- `cargo fmt --check` → exit 0
- `RAYON_NUM_THREADS=1 cargo test --features "training,env-bucket-brigade"` → all pass (417 in the main binary, 0 failed)
- Ran the probe end-to-end on the `beta05` cell (sample output above)

Closes #241

### CODE REVIEW

**Comment by rjwalters** [comment]
✅ **Approved!**

Thorough review against the #241 acceptance criteria — all four deliverables met, additive-only, no behavior change to existing trainers.

## Acceptance Criteria

1. **EV computed AND surfaced** — Verified `train_nfsp.rs` now prints `ev=` in the per-iter `br[...]` line, averaging `JointStats.explained_var` with the same `n` divisor as the existing `val`/`ent` diagnostics. The underlying value is `host_explained_variance` in `src/train/ppo/loss.rs:141` = `1 − Var(returns − values) / Var(returns)` (biased estimator, denom `n`), accumulated and epoch-averaged in `src/multi_agent/joint.rs:849,862`. Formula surfaced is exactly the one claimed.

2. **Standalone probe** — `examples/games/bucket_brigade/train_br_probe.rs` exists, runs on one cell, logs val-loss / EV / entropy / mean-return per iteration, env-configurable via `CELL`, `ITERATIONS`, `ROLLOUT_STEPS`, `VF_COEF`, `BR_TRAIN_STEPS`, `BR_REWARD_SCALE`. Confirmed it reuses existing plumbing (`collect_rollout`, `update_with_active_agents` with an active-mask selecting only `BR_AGENT`) rather than reinventing. Verified the freeze-N−1 invariant at `joint.rs:832-841`: only `active[i]` agents take an optimizer step; frozen agents drop gradients and are returned unchanged. No new training knobs — all six env vars map onto existing `JointTrainerConfig` fields. Nice detail: mean-return is accumulated from raw pre-scale rewards (before the `br_reward_scale` loop), so the logged value stays in native payoff units.

3. **One-line A/B command** — documented in both the PR body and the example's `//!` doc-comment.

4. **No behavior change** — Confirmed via merge-base diff: only `train_nfsp.rs` (additive log line), the new example, and `Cargo.toml` (example registration) are touched. Optimizer/update math untouched. `iterate_all_minibatches` already exists on main (prior #239/#240 work); the probe consumes it, doesn't introduce it. No scope creep.

## Test Execution

- `cargo fmt --check` → exit 0
- `cargo clippy --all-targets --features training -- -D warnings` → clean
- `cargo clippy --all-targets --features "training,env-bucket-brigade" -- -D warnings` → clean
- `cargo build --example train_br_probe --features "training,env-bucket-brigade"` → compiles
- `RAYON_NUM_THREADS=1 cargo test --lib --features "training,env-bucket-brigade"` → **417 passed, 0 failed, 3 ignored**

## CI

All checks pass (Tests Linux + macOS Burn NdArray, Clippy, Rustfmt, Documentation, Security Audit, Publish dry-run). `mergeStateStatus` = CLEAN.

Clean additive tooling. Good investigative ergonomics for the #239 BR-does-not-learn debugging.
